json/src/main.cpp

```diff
@@ -0,0 +1,178 @@
+                {
+                    printf("Child %d is an Array. Iterating though values.\n", i);
+                    const Array& arr = v.getArray();
+                    int j = 0;
+                    for (Array::iterator it = arr.begin(); it != arr.end(); it++) // Iterate over child values.
+                    {
+                        Value& va = *it;
+                        switch (va.getType())
+                        {
+                        case SCE_JSON_VALUE_BOOL:
+                        case SCE_JSON_VALUE_INT64:
+                        case SCE_JSON_VALUE_UINT64:
+                        case SCE_JSON_VALUE_REAL:
+                            va.toString(s);
+                            printf("Value %d: %s\n", j, s.c_str());
+                            break;
+                        case SCE_JSON_VALUE_STRING:
+                            printf("Value %d: %s\n", j, va.getString().c_str());
+                            break;
+                        default:
+                            printf("Value is null\n");
+                        }
+                        s.clear();
+                        ++j;
+                    }
+                }
+                break;
+            case SCE_JSON_VALUE_OBJECT: 
+                {
+                    printf("Child %d is an Object. Iterating though values.\n\n", i);
+                    const Object& obj = v.getObject();
+                    for (Object::Pair& pair : obj) // Iterate over child values.
+                    {
+                        printf("Key: %s\nType: %d\n", pair.key.c_str(), pair.value.getType());
+                        switch (pair.value.getType())
+                        {
+                        case SCE_JSON_VALUE_BOOL:
+                            printf("Value: %d\n", pair.value.getBoolean());
+                            break;
+                        case SCE_JSON_VALUE_INT64:
+                            printf("Value: %lld\n", (int)pair.value.getInteger());
+                            break;
+                        case SCE_JSON_VALUE_UINT64:
+                            printf("Value: %llu\n", pair.value.getUInteger());
+                            break;
+                        case SCE_JSON_VALUE_REAL:
+                            printf("Value: %f\n", pair.value.getReal());
+                            break;
+                        case SCE_JSON_VALUE_STRING:
+                            printf("Value: %s\n", pair.value.getString().c_str());
+                            break;
+                        default:
+                            printf("Value is null\n");
+                        }
+                    }
+                }
+                break;
+            default:
+                printf("Value is null");
+        }
+    }
+
+    String parsed = String();
+    val.serialize(parsed); // Serialize val into parsed. 
+    printf("\nSerialized Output Data: \n%s\n", parsed.c_str()); // Display parsed.
```

Comment:
Compare with the original using strcmp, usually this is how deserialization-serialization is checked with an autotest.